opal/core/log.py

```diff
@@ -18,10 +18,12 @@ def emit(self, record):
         record.args = []
         detail = ""
         if hasattr(record, "request") and record.request:
-            if record.request.user.is_authenticated():
-                user = record.request.user.username
-            else:
-                user = "anonymous"
+            user = "anonymous"
+            
+            if hasattr(record.request, "user"):
```

Comment:
Can we comment about the why - e.g. not blowing up if there's an error before the relevant middleware? 
I can entirely imagine me looking at this in 18 months and thinking this is superfluous...